fluxion/decoders/gstreamer.py

```diff
@@ -47,6 +50,20 @@ def decode(self, input_filepath: str, output_filepath: str):
                        stdout=subprocess.DEVNULL, check=True)
         return file_checksum(output_filepath)
 
+    @lru_cache
+    def check_run(self):
+        # pylint: disable=broad-except
+        elements = self.decoder_bin.split('!')
+        gst_inspect = f'gst-inspect-{self.gst_api}'
+        for element in elements:
+            element = element.strip().split()[0]
+            try:
+                subprocess.run([gst_inspect, element], stdout=subprocess.DEVNULL,
```

Comment:
You also need to check the exit code of the process, if gst-launch or gst-inspect is in the system but the decoder element isn't, the process will run correctly but it will exit with != 0

Reply:
The run call already has check=True as an argument. In case of exit code != 0, an exception is raised